server/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/RabbitMQWorkQueueTest.java

```diff
@@ -0,0 +1,122 @@
+    private static final Task TASK = new CompletedTask();
+    private static final Task TASK2 = new CompletedTask();
+    private static final TaskWithId TASK_WITH_ID = new TaskWithId(TASK_ID, TASK);
+    private static final TaskWithId TASK_WITH_ID_2 = new TaskWithId(TASK_ID_2, TASK2);
+
+    @RegisterExtension
+    static RabbitMQExtension rabbitMQExtension = RabbitMQExtension.singletonRabbitMQ();
+
+    private RabbitMQWorkQueue testee;
+    private TaskManagerWorker taskManagerWorker;
+    private JsonTaskSerializer taskSerializer;
+
+    @BeforeEach
+    void setUp() {
+        taskManagerWorker = mock(TaskManagerWorker.class);
+        when(taskManagerWorker.executeTask(TASK_WITH_ID)).thenReturn(Mono.just(Task.Result.COMPLETED));
+        when(taskManagerWorker.executeTask(TASK_WITH_ID_2)).thenReturn(Mono.just(Task.Result.COMPLETED));
+        taskSerializer = new JsonTaskSerializer(TestTaskDTOModules.COMPLETED_TASK_MODULE);
+        testee = new RabbitMQWorkQueue(taskManagerWorker, rabbitMQExtension.getRabbitConnectionPool(), taskSerializer);
+        testee.start();
+    }
+
+    @AfterEach
+    void tearDown() {
+        testee.close();
+    }
+
+    @Test
+    void workerShouldConsumeSubmittedTask() {
+        testee.submit(TASK_WITH_ID);
+
+        ArgumentCaptor<TaskWithId> taskWithIdCaptor = ArgumentCaptor.forClass(TaskWithId.class);
+        verify(taskManagerWorker, timeout(1000)).executeTask(taskWithIdCaptor.capture());
+
+        TaskWithId actualTaskWithId = taskWithIdCaptor.getValue();
+        assertThat(actualTaskWithId.getId()).isEqualTo(TASK_ID);
+        assertThat(actualTaskWithId.getTask().type()).isEqualTo(TASK.type());
+    }
+
+    @Test
+    void workerShouldConsumeTwoSubmittedTask() {
+        testee.submit(TASK_WITH_ID);
+        testee.submit(TASK_WITH_ID_2);
+
+        ArgumentCaptor<TaskWithId> taskWithIdCaptor = ArgumentCaptor.forClass(TaskWithId.class);
+        verify(taskManagerWorker, new Timeout(1000, times(2))).executeTask(taskWithIdCaptor.capture());
+
+        TaskWithId actualTaskWithId = taskWithIdCaptor.getAllValues().get(0);
+        assertThat(actualTaskWithId.getId()).isEqualTo(TASK_ID);
+        assertThat(actualTaskWithId.getTask().type()).isEqualTo(TASK.type());
+
+        TaskWithId actualSecondTaskWithId = taskWithIdCaptor.getAllValues().get(1);
+        assertThat(actualSecondTaskWithId.getId()).isEqualTo(TASK_ID_2);
+        assertThat(actualSecondTaskWithId.getTask().type()).isEqualTo(TASK2.type());
+    }
+
+    @Test
+    void givenTwoWorkQueuesOnlyTheFirstOneIsConsumingTasks() {
+        testee.submit(TASK_WITH_ID);
+
+        TaskManagerWorker otherTaskManagerWorker = mock(TaskManagerWorker.class);
+        RabbitMQWorkQueue otherWorkQueue = new RabbitMQWorkQueue(otherTaskManagerWorker, rabbitMQExtension.getRabbitConnectionPool(), taskSerializer);
+        otherWorkQueue.start();
+
+        for(int i = 0; i < 99; i++)
```

Comment:
You miss block quotes.

Please prefer an IntStream for that.

Also 100 seems overkill, please prefer 10...